bcgov/cas-registration
# Chore/267 task list activities

Adds the activities selected for a facility into the task list on the activity page.
**This PR refactors how we navigate to activities & how they are fetched from the backend**
- created a page.tsx file to replace all individual activity pages
- refactored the ActivityForm component to replace all individual activity components
- activities are fetched by their id rather than by name
- ~refactor form_builder service to get report_data from the facility_report~ Create a tech debt for this, out of scope here
- update activity page dev docs

![Screenshot from 2024-09-26 17-25-03](https://github.com/user-attachments/assets/d58a6d32-8731-4e5a-ae5f-67d0fa0feba3)


To test: 
- go to the reporting dashboard & create a report
- select activities in the facility review page (select ones that we have built)
- Go to /reporting/reports/{version_id}/facilities/{facility_id}/activities
- You should see the activities selected in the task list & be able to navigate to each by clicking on it
